src/const/dataSource.js

```diff
@@ -347,7 +349,9 @@ function getDataSourceCreationFields(dbmsTypeList, useOnlyVirtual = false, disab
         },
       },
     },
-  ] : [
+  ] : dbmsType === 'IMPALA' && !!useKerberos
```

Comment:
Imo, the combination of `? : && !! ? :` is not very readable. What do you think of moving the `credentialFields` into a separate function with regular `if`-s or `case`-s?